### PR TITLE
fix parallel make world

### PR DESCRIPTION
**Pull request by hendriktews**
fix two problems for make -j world; more problems remain for world.opt

### CODE REVIEW

**Comment by hendriktews** [comment]
This PR fixes two problems that I found with make -j 8 world. There are more bugs for a parallel build of world.opt.

**Comment by hendriktews** [comment]
I fixed the problem for make -j world.opt and squashed it into the existing patch.

**Review by damiendoligez — APPROVED** [review]
looks good

**Comment by mshinwell** [comment]
@damiendoligez Can we get this into 4.04?  Parallel make failures waste a fair amount of developer time in my experience.

File: ocamldoc/Makefile [hunk 1]
```diff
@@ -423,6 +423,6 @@ depend:
 	$(OCAMLLEX) odoc_lexer.mll
 	$(OCAMLLEX) odoc_ocamlhtml.mll
 	$(OCAMLLEX) odoc_see_lexer.mll
-	$(OCAMLDEP) $(INCLUDES_DEP) *.mll *.mly *.ml *.mli > .depend
+	$(OCAMLDEP) $(INCLUDES_DEP) *.mll *.mly *.ml *.mli generators/*ml > .depend
```
Comment (damiendoligez) [inline ocamldoc/Makefile]: This should be *.ml

**Comment by damiendoligez** [comment]
@mshinwell I'm OK with pushing this to 4.04.

**Comment by gasche** [comment]
Would it make sense to use a parallelization setting on some of the Travis CI machines to catch these issues in the commits where they arise?

File: ocamldoc/Makefile (hunk 1 above)
Comment (hendriktews) [reply ocamldoc/Makefile]: Fixed and changed.

**Comment by damiendoligez** [comment]
@gasche Yes it would make sense, in a new PR...

**Pull request merged**